crates/driver/src/domain/competition/solution/encoding.rs

```diff
@@ -230,19 +232,19 @@ pub fn tx(
         let flashloans = solution
             .flashloans
             .values()
-            .map(|flashloan| {
-                (
-                    flashloan.amount.0,
-                    flashloan.protocol_adapter.0,
-                    flashloan.liquidity_provider.0,
-                    flashloan.token.0.0,
-                )
+            .map(|flashloan| LoanRequest::Data {
+                amount: flashloan.amount.0.into_alloy(),
+                borrower: flashloan.protocol_adapter.0.into_alloy(),
+                lender: flashloan.liquidity_provider.0.into_alloy(),
+                token: flashloan.token.0.0.into_alloy(),
             })
             .collect();
 
-        let call = router.flash_loan_and_settle(flashloans, ethcontract::Bytes(settle_calldata));
-        let calldata = call.tx.data.unwrap().0;
-        (router.address().into(), calldata)
+        let calldata = router
+            .flashLoanAndSettle(flashloans, alloy::primitives::Bytes::from(settle_calldata))
```

Comment:
nit: probably no need to be so verbose here.
```suggestion
            .flashLoanAndSettle(flashloans, settle_calldata.into())
```